File: app/schemas/folder.py
from __future__ import annotations
import uuid
from typing import Optional, List
from datetime import datetime
from pydantic import BaseModel, ConfigDict


class FolderBase(BaseModel):
    workspace_id: uuid.UUID
    parent_folder_id: Optional[uuid.UUID] = None
    name: str
    folder_type: str = "custom"  # 'workflow', 'system', 'custom'
    workflow_stage_id: Optional[uuid.UUID] = None
    status: str = "active"


class FolderCreate(FolderBase):
    pass


class FolderUpdate(BaseModel):
    name: Optional[str] = None
    workflow_stage_id: Optional[uuid.UUID] = None
    status: Optional[str] = None


class FolderMoveRequest(BaseModel):
    new_parent_folder_id: Optional[uuid.UUID] = None


class FolderOut(FolderBase):
    id: uuid.UUID
    stage_name: Optional[str] = None
    stage_code: Optional[str] = None
    created_by: Optional[uuid.UUID] = None
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True)


class FolderTreeOut(FolderOut):
    subfolders: List[FolderTreeOut] = []

    model_config = ConfigDict(from_attributes=True)

File: frontend/src/types/folder.ts
// ============================================================
// Company OS — Folder Types
// ============================================================

export interface Folder {
  id: string;
  workspace_id: string;
  parent_folder_id?: string;
  name: string;
  folder_type: string;
  workflow_stage_id?: string;
  stage_name?: string;
  stage_code?: string;
  status: string;
  created_by?: string;
  created_at: string;
  updated_at: string;
}

export interface FolderTree extends Folder {
  subfolders: FolderTree[];
}

export interface FolderCreate {
  workspace_id: string;
  parent_folder_id?: string;
  name: string;
  folder_type?: string;
  workflow_stage_id?: string;
  status?: string;
}

export interface FolderUpdate {
  name?: string;
  workflow_stage_id?: string;
  status?: string;
}

export interface FolderMoveRequest {
  new_parent_folder_id?: string;
}
